devices/usbc/desc.c

```diff
@@ -0,0 +1,353 @@
+		desc_common.dc->status = DC_ADDRESS;
+
+		desc_common.dc->dev_addr = setup->wValue << 25;
+		desc_common.dc->dev_addr |= 1 << 24;
+
+		ctrl_setAddress(desc_common.dc->dev_addr);
+
+		desc_common.dc->op = DC_OP_INIT;
+		ctrl_execTransfer(0, (u32)desc_common.data->endpts[0].buf[USB_ENDPT_DIR_IN].buffer, 0, USB_ENDPT_DIR_IN);
+	}
+	else if (desc_common.dc->status != DC_CONFIGURED) {
+		desc_common.dc->status = DC_DEFAULT;
+	}
+}
+
+
+static void desc_ReqSetConfig(void)
+{
+	if (desc_common.dc->status == DC_ADDRESS) {
+		desc_common.dc->status = DC_CONFIGURED;
+		ctrl_execTransfer(0, (u32)desc_common.data->endpts[0].buf[USB_ENDPT_DIR_IN].buffer, 0, USB_ENDPT_DIR_IN);
+	}
+}
+
+
+static int desc_ReqGetConfig(const usb_setup_packet_t *setup)
+{
+	if (setup->wValue != 0 || setup->wIndex != 0 || setup->wLength != 1)
+		return ERR_NONE;
+
+	if (desc_common.dc->status != DC_CONFIGURED)
+		desc_common.data->endpts[0].buf[USB_ENDPT_DIR_OUT].buffer[0] = 0;
+	else
+		desc_common.data->endpts[0].buf[USB_ENDPT_DIR_OUT].buffer[1] = 1;
+
+	ctrl_execTransfer(0, (u32)desc_common.data->endpts[0].buf[USB_ENDPT_DIR_OUT].buffer, setup->wLength, USB_ENDPT_DIR_OUT);
+
+	return ERR_NONE;
+}
+
+
+static void desc_ReqGetDescriptor(const usb_setup_packet_t *setup)
+{
+	usb_string_desc_t *strdesc;
+
+	if (setup->wValue >> 8 == USB_DESC_DEVICE) {
+		ctrl_execTransfer(0, desc_common.dev, sizeof(usb_device_desc_t), USB_ENDPT_DIR_IN);
+	}
+	else if (setup->wValue >> 8 == USB_DESC_CONFIG) {
+		ctrl_execTransfer(0, desc_common.cfg, setup->wLength, USB_ENDPT_DIR_IN);
+	}
+	else if (setup->wValue >> 8 == USB_DESC_STRING) {
+		if ((setup->wValue & 0xff) == 0) {
+			strdesc = (usb_string_desc_t *)desc_common.str_0;
+			ctrl_execTransfer(0, desc_common.str_0, MIN(strdesc->bLength, setup->wLength), USB_ENDPT_DIR_IN);
+		}
+		else if ((setup->wValue & 0xff) == 1) {
+			strdesc = (usb_string_desc_t *)desc_common.str_man;
+			ctrl_execTransfer(0, desc_common.str_man, MIN(strdesc->bLength, setup->wLength), USB_ENDPT_DIR_IN);
+		}
+		else if ((setup->wValue & 0xff) == 2) {
+			strdesc = (usb_string_desc_t *)desc_common.str_prod;
+			ctrl_execTransfer(0, desc_common.str_prod, MIN(strdesc->bLength, setup->wLength), USB_ENDPT_DIR_IN);
+		}
+		else if ((setup->wValue & 0xff) == 4) {
+			ctrl_execTransfer(0, (u32)desc_common.data->endpts[0].buf[USB_ENDPT_DIR_IN].buffer, 0, USB_ENDPT_DIR_IN);
+			ctrl_execTransfer(0, (u32)desc_common.data->endpts[0].buf[USB_ENDPT_DIR_OUT].buffer, 71, USB_ENDPT_DIR_OUT);
+			return;
+		}
+	}
+	else if (setup->wValue >> 8 == USB_DESC_TYPE_HID_REPORT) {
+		ctrl_execTransfer(0, desc_common.hid_reports, 76, USB_ENDPT_DIR_IN);
+	}
+
+	ctrl_execTransfer(0, (u32)desc_common.data->endpts[0].buf[USB_ENDPT_DIR_OUT].buffer, 0x40, USB_ENDPT_DIR_OUT);
+}
+
+
+static void desc_defaultSetup(const usb_setup_packet_t *setup)
+{
+	u32 fsz;
+
+	if (*(u32 *)setup == 0xdeadc0de) {
+		desc_common.dc->op = DC_OP_EXIT;
+	}
+	else {
+		fsz = setup->wValue << 16;
+		fsz |= setup->wIndex;
```

Comment:
`fsz` is assigned  a value that is never used.

Reply:
Thanks, it was removed.